builder/src/lib.rs

```diff
@@ -183,17 +199,8 @@ pub fn build_firmware_elfs_uncached<'a>(
         }
         run_cmd(&mut cmd)?;
 
-        let target_dir = if let Some(dir) = std::env::var_os("CARGO_TARGET_DIR") {
-            PathBuf::from(dir)
-        } else {
-            Path::new(workspace_dir).join("target")
-        };
-
         for &fwid in invocation.fwids.iter() {
-            result_map.insert(
-                fwid,
-                fs::read(target_dir.join(TARGET).join(PROFILE).join(fwid.bin_name))?,
-            );
+            result_map.insert(fwid, fs::read(get_elf_path(fwid).unwrap())?);
```

Comment:
Can you explain a bit more about what's going on here? I don't think we want to look at CALIPTRA_PREBUILT_FW_DIR when we're building; that's likely to cause trouble with some of my shell scripts.

Reply:
I think I see the problem now. I had _assumed_ that this would only be run in the non-CALIPTRA_PREBUILT_FW_DIR case, not checking for the other callers. I modified this function to call get_elf_path to avoid having two places where the path is constructed. I'm debating reverting this modification or trying to do it in a different way.